src/server/cluster/cluster_family.cc

```diff
@@ -281,8 +281,9 @@ void ClusterFamily::DflyCluster(CmdArgList args, ConnectionContext* cntx) {
     return (*cntx)->SendError("DFLYCLUSTER commands requires --cluster_mode=yes");
   }
   CHECK_NE(cluster_config_.get(), nullptr);
-
-  // TODO check admin port
+  if (!(cntx->is_test) && !cntx->owner()->IsAdmin()) {
```

Comment:
Here's an idea:

1. We could introduce a `RunAdmin()` method in `BaseFamilyTest`, in addition to existing `Run()`
2. This `RunAdmin()` would tweak the `TestConnection` to respond to `IsAdmin()` with `true` (we can either make `IsAdmin()` virtual, or modify the port somehow)

If we'll do this, we gain the benefits of
1. Not having to modify production code with hooks
2. Being able to test the behavior of rejecting non-admin commands via non-admin ports

wdyt?

Reply:
Sure I will work on making IsAdmin virtual